control_integration_test.go

```diff
@@ -0,0 +1,71 @@
+//go:build integration && scylla
+// +build integration,scylla
+
+package gocql
+
+import (
+	"context"
+	"fmt"
+	"net"
+	"testing"
+)
+
+// unixSocketDialer is a special dialer which connects only to the maintenance_socket.
+type unixSocketDialer struct {
+	dialer     net.Dialer
+	socketPath string
+}
+
+func (d unixSocketDialer) DialContext(_ context.Context, _, _ string) (net.Conn, error) {
+	return d.dialer.Dial("unix", d.socketPath)
+}
+
+func TestUnixSockets(t *testing.T) {
+	socketPath := "/tmp/scylla/cql.m"
+
```

Comment:
How does this socket gets opened at that path?

Reply:
socket is open in workdir (`maintenance_socket: workdir`) in the container which is by default `/var/lib/scylla/` but I map it to the `/tmp/scylla` in `docker-compose.yml`